src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.Client/ProjectProperties/ProjectPropertiesEditorFactory.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.VisualStudio.ProjectSystem.VS;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using IServiceProvider = Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
+
+namespace Microsoft.VisualStudio.Packaging
+{
+    [Guid(EditorFactoryGuidString)]
+    [ProvideView(LogicalView.Designer, "Design")]
+    internal sealed class ProjectPropertiesEditorFactory : IVsEditorFactory
+    {
+        private const string EditorFactoryGuidString = "04B8AB82-A572-4FEF-95CE-5222444B6B64";
+
+        // TODO if the new editor is not enabled, fall back to the legacy editor via this GUID
+        private const string LegacyEditorFactoryGuidString = "990036EB-F67A-4B8A-93D4-4663DB2A1033";
+
+        /// <summary>
+        /// Logical view identifier (passed to <see cref="MapLogicalView"/>) when the properties
+        /// pages are to be launched displaying the "debug" page.
+        /// </summary>
+        /// <remarks>
+        /// Triggered via menu item "Debug | [Project Name] Debug Properties".
+        /// </remarks>
+        private static readonly Guid DebugPageLogicalViewGuid = new Guid("0273C280-1882-4ED0-9308-52914672E3AA");
```

Comment:
>DebugPageLogicalViewGuid [](start = 37, length = 24)

Please remind me, what makes the debug page different?

Reply:
The debug pages has behavior that doesn't really belong to the properties pages. It has the ability to create profiles for debugging which get saved to launchsettings.json

Reply:
The idea here is that you may want to open the project property pages directly to the Debug page. I can't recall if we ever supported opening directly to the 

Whether the debug settings should even live in the project property pages is a debatable point. :-) They are logically project-level settings; they just happen to be persisted in launchSettings.json rather than an MSBuild file.

Reply:
In this case, the debug page is treated separately because there's a feature that navigates directly to it.

> Whether the debug settings should even live in the project property pages is a debatable point. :-) They are logically project-level settings; they just happen to be persisted in launchSettings.json rather than an MSBuild file.

To expand upon this a bit, the debug page has a few differences to the other pages:

- There are logically multiple debug pages, one per launch profile, although as the user it's hard to see this (a dropdown controls this, which is unique to this page):
   ![image](https://user-images.githubusercontent.com/350947/88012889-6203c500-cb5e-11ea-8be4-a1153218f8d5.png)
   This adds a lot of complexity to the code today, as it doesn't really fit the model of the other pages.
- Configurations (debug/release, platform) do not apply to them.
- It makes sense for a user to want to not only navigate directly to the launch profile editor UI, but to a specific launch profile (which is currently not supported). For example via this toolbar button:
   ![image](https://user-images.githubusercontent.com/350947/88012973-9bd4cb80-cb5e-11ea-9f5e-c4eb456aba8b.png)

- They are stored in a separate file (though that could change).
- They are usually not added to source control (though we do, and there is a use case for doing so).

Breaking them out to a new UI would allow us create an experience that better matches the nature of these launch profiles, and makes it easier to manage them. It'd also likely have a different extensibility model than the other property pages, because the options available depend upon the specific provider.

It's worth noting that we would be able to re-use much of the code between project properties and launch profiles.